### PR TITLE
Create .bak file when loading fails

### PR DESCRIPTION
**Pull request by yesseruser**
When loading a file that has errors in it, Bloxstrap would previously overwrite the file with the defaults. This PR copies the file to a .bak file with the same name so the user can choose to edit it and fix issues.

Related to #3491.

### CODE REVIEW

**Comment by MrT-tak** [comment]
Love it!
Makes my comment true after Pizzaboxer werE workinG on it but left.

File: Bloxstrap/JsonManager.cs [hunk 1]
```diff
@@ -47,6 +47,9 @@ public virtual void Load(bool alertFailure = true)
 
                     if (!String.IsNullOrEmpty(message))
                         Frontend.ShowMessageBox($"{message}\n\n{ex.Message}", System.Windows.MessageBoxImage.Warning);
+                    
+                    // Create a backup of loaded file
+                    File.Copy(FileLocation, FileLocation + ".bak");
```
Comment (bluepilledgreat) [inline Bloxstrap/JsonManager.cs:52]: This will fail if a backup file already exists.

**Review by bluepilledgreat — CHANGES_REQUESTED** [review]
can you address what i commented?

**Comment by MrT-tak** [comment]
> can you address what i commented?

I think @yesseruser can do it. If people tell me, I know Javascript so I can try to help!

**Comment by MrT-tak** [comment]
I'll try. My best...

**Comment by MrT-tak** [comment]
@bluepilledgreat What do you want to do when it fails?

File: Bloxstrap/JsonManager.cs (hunk 1 above)
Comment (MrT-tak) [reply Bloxstrap/JsonManager.cs:52]: As I wrote in pr, what should the behaviour be when it fails?

**Comment by MrT-tak** [comment]
Not the best code but I have something!

**Comment by MrT-tak** [comment]
I'm a newbie to PRs, sorry!
This is what I did
```cs
// Create a backup of loaded file
                    try
                    {
                        File.Copy(FileLocation, FileLocation + ".bak");
                    }
                    catch
                    {
                        App.Logger.WriteLine(LOG_IDENT, "Backup file attempted to be created but failed. Please create a issue.");
                    }
```

**Comment by yesseruser** [comment]
> I'm a newbie to PRs, sorry!
> This is what I did
> ```cs
> // Create a backup of loaded file
>                     try
>                     {
>                         File.Copy(FileLocation, FileLocation + ".bak");
>                     }
>                     catch
>                     {
>                         App.Logger.WriteLine(LOG_IDENT, "Backup file attempted to be created but failed. Please create a issue.");
>                     }
> ```
> 
> I don't have any thing to test.
> 

I don't think that an issue needs to be created when it fails. If the file already exists, that means a backup was created before so I think the app should ask the user if they want to replace the file or drop the new backup. Or we could just add something like .2.bak if they do overlap. Opinions?

**Comment by bluepilledgreat** [comment]
> I don't think that an issue needs to be created when it fails. If the file already exists, that means a backup was created before so I think the app should ask the user if they want to replace the file or drop the new backup. Or we could just add something like .2.bak if they do overlap. Opinions? @bluepilledgreat

Just delete the previous backup (if one exists). Don't quite like the idea of flooding the user's folder with countless backup files. Also a prompt for this is stupid.

**Comment by yesseruser** [comment]
> > I don't think that an issue needs to be created when it fails. If the file already exists, that means a backup was created before so I think the app should ask the user if they want to replace the file or drop the new backup. Or we could just add something like .2.bak if they do overlap. Opinions? @bluepilledgreat
> 
> Just delete the previous backup (if one exists). Don't quite like the idea of flooding the user's folder with countless backup files. Also a prompt for this is stupid.

Ok, I'll work on that when I get home.

**Comment by MrT-tak** [comment]
> > > I don't think that an issue needs to be created when it fails. If the file already exists, that means a backup was created before so I think the app should ask the user if they want to replace the file or drop the new backup. Or we could just add something like .2.bak if they do overlap. Opinions? @bluepilledgreat
> > 
> > 
> > Just delete the previous backup (if one exists). Don't quite like the idea of flooding the user's folder with countless backup files. Also a prompt for this is stupid.
> 
> Ok, I'll work on that when I get home.

Thanks!

**Review by bluepilledgreat — APPROVED** [review]

**Comment by MrT-tak** [comment]
Woohooo!

**Pull request merged**